src/media/track.ts

```diff
@@ -0,0 +1,220 @@
+     */
+    this.#mediaStreamTrack.onended = () => {
+      this.emit(Events.Ended, { trackState: this.trackState });
+    };
+
+    /**
+     * Emit mute event when the underlying track gets muted.
+     */
+    this.#mediaStreamTrack.onmute = () => {
+      // using arrow function which should bind to this from outer scope track
+      const action = this.#mediaStreamTrack.enabled ? 'muted' : 'unmuted';
+      // TODO:  Move this logic else where
+
+      this.emit(Events.Muted, {
+        action,
+      });
+    };
+  }
+
+  /**
+   * Get id of this track.
+   *
+   * @returns The id of this track.
+   */
+  get id(): string {
+    return this.#mediaStreamTrack.id;
+  }
+
+  /**
+   * Get muted state of this track.
+   *
+   * @returns The muted state of this track.
+   */
+  get muted(): boolean {
+    return !this.#mediaStreamTrack.enabled;
+  }
+
+  /**
+   * Get current state of this track.
+   *
+   * @returns Current state of this track.
+   */
+  get trackState(): TrackState {
+    return { id: this.id, label: this.label, muted: this.muted };
+  }
+
+  /**
+   * Set the mute state of this track.
+   *
+   * @param muted - True to mute, false to unmute.
+   * @fires Events.Muted
+   */
+  setMuted(muted: boolean): void {
+    // Only change state if it's different, where "enabled" means "unmuted."
+    if (this.#mediaStreamTrack.enabled === muted) {
+      this.#mediaStreamTrack.enabled = !muted;
+      this.emit(Events.Muted, { trackState: this.trackState });
+      logger.log(`Local track ${muted ? 'muted' : 'unmuted'}:`, { trackState: this.trackState });
+    }
+  }
+
+  /**
+   *
+   * This method returns the actual MediaStreamTrack.
+   *
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  getMediaStreamTrack(): MediaStreamTrack {
+    return this.#mediaStreamTrack;
+  }
+
+  /**
+   *
+   * This method returns track after applying effects.
+   *
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  getMediaStreamTrackWithEffects(): MediaStreamTrack {
+    return this.#mediaStreamTrackWithEffect;
+  }
+
+  /**
+   *
+   * This method sets the track after applying effect.
+   *
+   * @param track
+   * @returns #mediaStreamTrack of type MediaStreamTrack.
+   */
+  setMediaStreamTrackWithEffects(track: MediaStreamTrack): void {
+    this.#mediaStreamTrackWithEffect = track;
+    this.emit(Events.UnderlyingTrackChange);
+  }
+
+  /**
+   * Stop this track.
+   *
+   * @fires Events.Ended
+   */
+  stop(): void {
+    this.#mediaStreamTrack.stop();
+    this.emit(Events.Ended, { trackState: this.trackState });
+    logger.log(`Local track stopped:`, { trackState: this.trackState });
+  }
+
+  /**
+   *
+   * @param element - Either pass the ID for the dom element where we need to attach the video tag
+   * pass the video element dom element where we need to attach the stream.
+   */
+
+  /**
+   * @param element
+   */
+  play(element?: string | HTMLElement): void {
```

Comment:
Does this need to be part of the track API?  I thought we were leaving this to the user.

Reply:
This would be really nice to have so that we can keep track on the dom element where the audio or video is playing 

mainly in the mobile device we can monitor the status of audio and video getting played for metrics and also see if its working properly , we could also play if underlying track changes 

its is more of a helper else developer have to write there own custom logic

Reply:
Could it be done at a higher level then? It's not something that feels like belongs in webrtc-core.

Reply:
i can add them on just remote track later

Reply:
they should definitely be on remote track if we're going to keep this api, but i'm talking about moving this functionality out of webrtc-core altogether

Reply:
When thinking about it , i might have to extend all the classes like camera , microphone , display methods

let me see if there is any other way i can introduce this function

Reply:
removed it